### PR TITLE
docs: add detail pane docs

### PR DESCRIPTION
**Pull request by gnomeontherun**
This provides the detail pane documentation, as well as a small bugfix for the column toggle icon and forcing the detail pane widths to 1/3 | 2/3.

## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [ ] No

### CODE REVIEW

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Detail Pane</h2>
+
+<p>
+    When you need to show additional details for a record, the Detail Pane is a good pattern that will condense the datagrid to show a primary column and a panel to the right that you can display more details about your record. It is a great pattern to show the full content of the record in a larger space that can also scroll. The Detail Pane is also fully accessible for keyboard and screen reader users.
+</p>
+<p>
+    Adding the Detail Pane will add a new toggle icon in the left hand side of the Datagrid. When the pane is opened, it will take up 2/3 of the width of the Datagrid and automatically hide all columns, except for the first column and any built-in columns that facilitate features like selection and row actions. The pagination is also changed to a condensed format that scales well to small sizes. Only one row can be opened at a time, so selecting another row will change the content to the newly selected row.
```
Comment (coryrylan) [inline src/website/src/app/documentation/demos/datagrid/detail/detail.html]: Update to present tense and active voice

"The Detail Pane adds a new toggle icon on the left-hand side of the Datagrid. When the pane is open, it takes 2/3 of the width of the Datagrid and hides all columns, except for the first column and any built-in columns that facilitate features like selection and row actions. The pagination also updates to a condensed format that scales well to small sizes. Only one row can be open at a time; selecting another row changes the content to the newly selected row."

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+</p>
+<p>
+    The Detail Pane cannot be used at the same time as Expandable Rows, and if both are enabled at the same time the Detail Pane will take presedence. Also, hide and show columns is disabled while the Detail Pane is opened, but stil works properly when closed. The rest of the Datagrid behaviors should work as expected even while the Detail Pane is opened.
```
Comment (coryrylan) [inline src/website/src/app/documentation/demos/datagrid/detail/detail.html]: Minor spelling, active voice and present tense 

"The Detail Pane is not compatible with Expandable Rows; when both are enabled, the Detail Pane takes precedence. Hide and show columns are disabled while the Detail Pane is open, but still works properly when closed. The rest of the Datagrid behaviors work as expected, even while the Detail Pane is open."

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    <clr-dg-row *clrDgItems="let user of users" [clrDgItem]="user">
+        <clr-dg-cell>{{user.id}}</clr-dg-cell>
+        <clr-dg-cell>{{user.name}}</clr-dg-cell>
+        <clr-dg-cell>{{user.creation | date}}</clr-dg-cell>
+        <clr-dg-cell>
+            <span class="color-square" [style.backgroundColor]="user.color"></span>
+        </clr-dg-cell>
+    </clr-dg-row>
+
+    <clr-dg-detail *clrIfDetail="let detail" style="width: 30%">
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datagrid/detail/detail.html]: fixed

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+<p>
+    Adding the Detail Pane will add a new toggle icon in the left hand side of the Datagrid. When the pane is opened, it will take up 2/3 of the width of the Datagrid and automatically hide all columns, except for the first column and any built-in columns that facilitate features like selection and row actions. The pagination is also changed to a condensed format that scales well to small sizes. Only one row can be opened at a time, so selecting another row will change the content to the newly selected row.
+</p>
+<p>
+    The Detail Pane cannot be used at the same time as Expandable Rows, and if both are enabled at the same time the Detail Pane will take presedence. Also, hide and show columns is disabled while the Detail Pane is opened, but stil works properly when closed. The rest of the Datagrid behaviors should work as expected even while the Detail Pane is opened. 
+</p>
+
+<h3>Basic use of Detail Pane</h3>
+
+<p>To use the Detail Pane, you simply add a new element with the following syntax inside of the Datagrid</p>
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datagrid/detail/detail.html]: fixed

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html [hunk 5]
```diff
@@ -0,0 +1,117 @@
+<!--
+  ~ Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>Detail Pane</h2>
+
+<p>
+    When you need to show additional details for a record, the Detail Pane is a good pattern that will condense the datagrid to show a primary column and a panel to the right that you can display more details about your record. It is a great pattern to show the full content of the record in a larger space that can also scroll. The Detail Pane is also fully accessible for keyboard and screen reader users.
```
[the remark this answers is not in the document]
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datagrid/detail/detail.html]: fixed

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html (hunk 1 above)
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datagrid/detail/detail.html]: fixed

File: src/website/src/app/documentation/demos/datagrid/detail/detail.html (hunk 2 above)
Comment (gnomeontherun) [reply src/website/src/app/documentation/demos/datagrid/detail/detail.html]: fixed

**Pull request merged**